### PR TITLE
Cursor cleanup - part 1

### PR DESCRIPTION
**Pull request by JJones780**
Recent cursor changes revealed growing pains related to the new input system.
This keeps the new cursor behaviour ( but consolidates the responsibility in XournalpppCursor ) and re-enables functionality that hadn't yet carried over to the new input system.
It also retains new and old functionality in the old input system.

### CODE REVIEW

File: src/gui/XournalppCursor.cpp [hunk 1]
```diff
@@ -118,119 +132,127 @@ void XournalppCursor::updateCursor()
...
 
 	GdkCursor* cursor = NULL;
-
+	bool useDefault = false;
+	
+	
 	if (this->busy)
 	{
-		cursor = gdk_cursor_new_for_display(gdk_window_get_display(window), GDK_WATCH);
+		setStockCursor(GDK_WATCH);
```
Comment (Technius) [inline src/gui/XournalppCursor.cpp]: The combination of `setStockCursor` and `cursor = ...` is error prone. I suggest refactoring this so that only one of `setStockCursor` or `cursor = ...` is used.

File: src/gui/XournalppCursor.h [hunk 2]
```diff
@@ -32,8 +33,9 @@ class XournalppCursor
 	void setMouseDown(bool mouseDown);
 	void setInvisible(bool invisible);
 	void setInsidePage(bool insidePage);
-	void setTempCursor(GdkCursorType type);
-	void setTempDrawDirCursor(bool shift, bool ctrl);
+	void setStockCursor(GdkCursorType type);
```
Comment (Technius) [inline src/gui/XournalppCursor.h]: Document what `setStockCursor` does, the purpose is unclear right now.

File: src/gui/XournalppCursor.h [hunk 3]
```diff
@@ -57,4 +63,14 @@ class XournalppCursor
...
 	bool invisible = false;
+	
+	// One shot drawDir custom cursor -drawn instead of pen/stylus then cleared.
+	bool drawDirActive = false;
+	bool drawDirShift = false;
+	bool drawDirCtrl = false;	
+	
+	
+	//combination to avoid making same cursor
+	void* lastCustomCursorAddress = NULL;	//for comparison only
```
Comment (Technius) [inline src/gui/XournalppCursor.h]: This doesn't seem to be used anywhere.

File: src/gui/XournalppCursor.cpp [hunk 4]
```diff
@@ -118,119 +132,127 @@ void XournalppCursor::updateCursor()
...
+			{
+				setStockCursor(GDK_ARROW);
+			}
+			else
+			{
+				if (type == TOOL_PEN)
+				{
+					cursor = getPenCursor();
+				}
+				else // must be:  if (type == TOOL_HILIGHTER)
```
Comment (Technius) [inline src/gui/XournalppCursor.cpp]: Either add an assertion here or keep the highlighter tool check; this assumption may not be true in the future.

File: src/gui/inputdevices/MouseInputHandler.cpp [hunk 5]
```diff
@@ -54,19 +54,12 @@ bool MouseInputHandler::handleImpl(InputEvent* event)
 	 * Trigger motion actions
 	 */
 	// Trigger motion action when pen/mouse is pressed and moved
-	if (event->type == MOTION_EVENT) //mouse or pen moved
+	if ( event->type == MOTION_EVENT) //mouse or pen moved
```
Comment (Technius) [inline src/gui/inputdevices/MouseInputHandler.cpp]: Unnecessary whitespace.

File: src/gui/XournalppCursor.cpp [hunk 6]
```diff
@@ -254,8 +275,10 @@ void XournalppCursor::updateCursor()
 		{
 			gdk_window_set_cursor(gtk_widget_get_window(xournal->getWidget()), cursor);
 		}
-
-		gtk_widget_set_sensitive(xournal->getWidget(), !this->busy);
+		else if(useDefault)
```
Comment (Technius) [inline src/gui/XournalppCursor.cpp]: `useDefault` should be renamed, since it does not "use a default cursor" but rather disables inputs.

File: src/gui/XournalppCursor.cpp [hunk 7]
```diff
@@ -118,119 +132,127 @@ void XournalppCursor::updateCursor()
...
 			case CURSOR_SELECTION_BOTTOM:
-				cursor = gdk_cursor_new_for_display(gdk_window_get_display(window), GDK_SB_V_DOUBLE_ARROW);
+				setStockCursor(GDK_SB_V_DOUBLE_ARROW);
 				break;
 			default:
 				break;
 			}
-		}
-		else if (type == TOOL_PEN)
+		}
```
Comment (Technius) [inline src/gui/XournalppCursor.cpp]: Trailing whitespace.

File: src/gui/XournalppCursor.cpp [hunk 8]
```diff
@@ -302,16 +325,32 @@ GdkCursor* XournalppCursor::getEraserCursor()
 GdkCursor* XournalppCursor::getHighlighterCursor()
 {
 	XOJ_CHECK_TYPE(XournalppCursor);
+	
+	if (this->drawDirActive)
+	{
+		this->drawDirActive = false;
```
Comment (Technius) [inline src/gui/XournalppCursor.cpp]: This is error prone because it modifies the state: updating the cursor twice a row will not create the correct cursor.

File: src/gui/XournalppCursor.h [hunk 9]
```diff
@@ -57,4 +63,14 @@ class XournalppCursor
 
 	bool mouseDown = false;
 	bool invisible = false;
+	
+	// One shot drawDir custom cursor -drawn instead of pen/stylus then cleared.
+	bool drawDirActive = false;
+	bool drawDirShift = false;
+	bool drawDirCtrl = false;
```
Comment (Technius) [inline src/gui/XournalppCursor.h]: Please leave comments explaining how the three are related.

**Comment by JJones780** [comment]
@Technius,  Thanks for having a look. Sorry I didn't put a comment in to mention I was rewriting the XournalppCursor class next.
